panel/panel.py

```diff
@@ -207,107 +185,117 @@ def fill_panel(window, update=False, **panel_filter):
     else:
         settings.set("sublime_linter_panel_filter", panel_filter)
 
-    to_render = RenderLines()
-    for vid, view_dict in errors.items():
-        if util.is_none_or_zero(view_dict["we_count_view"]):
+    to_render = []
+    for bid, buf_errors in errors_by_bid.items():
+        # do not show headers without errors
+        if not buf_errors:
             continue
 
-        view_lines = get_view_lines(view_dict, panel_filter)
-        if view_lines:
-            # append header
-            to_render.append(format_header(path_dict[vid]))
+        # append header
+        to_render.append(format_header(path_dict[bid]))
 
-            # append lines
-            for lineno, error_type, item in view_lines:
-                to_render.append(format_row(lineno, error_type, item))
-                item["panel_lineno"] = to_render.current_lineno()
+        # append lines
+        base_lineno = len(to_render)
+        for i, item in enumerate(buf_errors):
+            to_render.append(format_row(item))
+            item["panel_line"] = base_lineno + i
 
-            # insert empty line between views sections
-            to_render.append("")
+        # insert empty line between views sections
+        to_render.append("")
 
-    rendered_text = to_render.render() if to_render else None
-    run_update_panel_cmd(panel, text=rendered_text)
+    run_update_panel_cmd(panel, text="\n".join(to_render))
 
 
 # logic for updating panel selection
 
-def get_closest_region_dict(dic, colno):
-    dics = [
-        d
-        for error_dict in dic.values() for d in error_dict
-        if d.get("panel_lineno")
-        # if d["start"] <= colno <= d["end"]  # problematic line
-    ]
-    if not dics:
-        return
-    return min(dics, key=lambda x: abs(x["start"] - colno))
+def get_next_panel_line(buf_line, buf_errors):
+    """Get panel line for next error in buffer.
 
-
-def get_next_lineno(num, interval):
-    interval = set(interval)
-    interval.discard(num)
-    interval = list(interval)
-    interval.sort()
-
-    if num < interval[0] or interval[-1] < num:
-        return interval[0]
+    If not found this means the current line is past last error's buffer line.
+    In that case return last error's panel line incremented by one, which will
+    place panel selection in empty space between buffer sections.
+    """
+    for d in buf_errors:
```

Comment:
d -> error

Reply:
I should’ve caught this too. Unless a var is short lived, eg. scopes inside a short loop or something, it shouldn’t have a single character for a name.